frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.cyspec.ts

```diff
@@ -0,0 +1,190 @@
+    .set('persistentvolumeclaims', { kind: 'PersistentVolumeClaim' })
+    .set('statefulsets', { kind: 'StatefulSet' })
+    .set('resourcequotas', { kind: 'ResourceQuota' })
+    .set('limitranges', { kind: 'LimitRange' })
+    .set('horizontalpodautoscalers', { kind: 'HorizontalPodAutoscaler' })
+    .set('networkpolicies', { kind: 'NetworkPolicy' })
+    .set('roles', { kind: 'Role' });
+  const openshiftObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>()
+    .set('deploymentconfigs', { kind: 'DeploymentConfig' })
+    .set('buildconfigs', { kind: 'BuildConfig' })
+    .set('imagestreams', { kind: 'ImageStream' })
+    .set('routes', { kind: 'Route' })
+    .set('user.openshift.io~v1~Group', { kind: 'user.openshift.io~v1~Group', namespaced: false });
+  const serviceCatalogObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>().set(
+    'clusterservicebrokers',
+    {
+      kind: 'servicecatalog.k8s.io~v1beta1~ClusterServiceBroker',
+      namespaced: false,
+    },
+  );
+
+  let testObjs = Cypress.env('openshift') === true ? k8sObjs.merge(openshiftObjs) : k8sObjs;
+  testObjs = Cypress.env('servicecatalog') === true ? testObjs.merge(serviceCatalogObjs) : testObjs;
+  const testLabel = 'automatedTestName';
+  const resourcesWithCreationForm = new Set(['StorageClass', 'Route', 'PersistentVolumeClaim']);
+  const leakedResources = new Set<string>();
+
+  testObjs.forEach(({ kind, namespaced = true }, resource) => {
+    describe(kind, () => {
+      const name = `${testName}-${_.kebabCase(kind)}`; // end name with '&' to force test failures
+      it('displays a list view for the resource', () => {
+        cy.visit(
+          `${namespaced ? `/k8s/ns/${testName}` : '/k8s/cluster'}/${resource}?name=${testName}`,
+        );
+      });
+      if (namespaced) {
+        it('has a working namespace dropdown on namespaced objects', () => {
+          listPage.projectDropdownShouldExist();
+          listPage.projectDropdownShouldContain(testName);
+        });
+      } else {
+        it('does not have a namespace dropdown on non-namespaced objects', () => {
+          listPage.projectDropdownShouldNotExist();
+        });
+      }
+      it('creates, searches for, edits, and deletes a resource instance', () => {
+        listPage.clickCreateYAMLButton();
+        if (resourcesWithCreationForm.has(kind)) {
+          cy.get('#yaml-link').click();
+        }
+        // for 'create yaml' resource sidebar is rendered last
+        cy.byTestID('resource-sidebar').should('exist');
+        let newContent;
+        // get, update, and set yaml editor content.
+        yamlEditor
+          .getEditorContent()
+          .then((content) => {
+            newContent = _.defaultsDeep(
+              {},
+              { metadata: { name, labels: { [testLabel]: testName } } },
+              safeLoad(content),
+            );
+          })
+          .then(() => {
```

Comment:
You shouldn't need to chain a promise here since the function above doesn't return a promsise

Reply:
but I want it run after the first 'then'

Reply:
> but I want it run after the first 'then'

Just make it part of the first then

Reply:
It would still need to be a nested 'then'.  To me it just reads easier -easier to see the 2nd then as opposed to it being nested within the first 'then'.

Reply:
need to `getEditorContent()` then do `newContent`, then `setEditorContent()`, then click `Save` button.   I prefer `foo().then().then()` over `foo.then( fooB.then() )`

Reply:
Why do you need another `then`? I don't see a second promise. If you don't have a return value from `then`, it won't wait for any promise to resolve.

Reply:
Hi @spadgett I believe if I don't have a second `then`, then `newContent` will not be set since it is being set  in the first `then`.   In other words if I set `newContent` in first `then` following the `getEditorContent`.  If `setEditorContent` is not in a 'then` it will execute before the first `then` and `newContent` will not be set.

Reply:
Let's step back. I'm proposing that instead of this:

```ts
.then(() => {
  foo();
})
.then(() => {
  bar();
});
```

You should do this:

```ts
.then(() => {
  foo();
  bar();
});
```

Reply:
with `bar()` being `setEditorContent()` which uses `cy.window` which is a promise.

Reply:
> with `bar()` being `setEditorContent()` which uses `cy.window` which is a promise.

That doesn't actually change anything. It only matters if the first statement `foo()` is a promise.

Reply:
This would be simpler with `async`/`await`, although I've hesitated to recommend that as it can cause us to lose the full stack trace on failures (at least with Protractor).

Reply:
I just spun up a cluster so ready to start on this again, so wouldn't it have to be:
```
.then(() => {
  foo();
  bar().then(...);
});
```

Reply:
@dtaylor113 If you want to block on `bar()`, I'd recommend:

```ts
.then(() => {
  foo();
  return bar();
}).then(() => {
  // do stuff
});
```

Note that everything after this point in the test needs to be where `// do stuff` is.

async/await would be much simpler here.

```ts
await firstRequest();
foo();
await bar();
// do stuff
```

Reply:
fixed